dask_image/ndmeasure/_utils/_find_objects.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import pandas as pd
+from dask.delayed import delayed, Delayed
+import dask.dataframe as dd
+
+
+def _array_chunk_location(block_id, chunks):
+    """Pixel coordinate of top left corner of the array chunk."""
+    array_location = []
+    for idx, chunk in zip(block_id, chunks):
+        array_location.append(sum(chunk[:idx]))
+    return tuple(array_location)
+
+
+@delayed
+def _find_bounding_boxes(x, array_location):
+    """An alternative to scipy.ndi.find_objects"""
+    unique_vals = np.unique(x)
+    unique_vals = unique_vals[unique_vals != 0]
+    result = {}
+    for val in unique_vals:
+        positions = np.where(x == val)
+        slices = tuple(slice(np.min(pos) + array_location[i], np.max(pos) + 1 + array_location[i], 1) for i, pos in enumerate(positions))
+        result[val] = slices
+    return pd.DataFrame.from_dict(result, orient='index')
+
+
+def _combine_slices(slices):
+    "Return the union of all slices."
+    if len(slices) == 1:
+        return slices[0]
+    else:
+        start = min([sl.start for sl in slices])
+        stop = max([sl.stop for sl in slices])
+        return slice(start, stop, 1)
+
+
+def _merge_bounding_boxes(x, ndim):
+    x = x.dropna()
+    data = {}
+    for i in range(ndim):
+        slices = [x[ii] for ii in x.index if str(ii).startswith(str(i))]
```

Comment:
I'm a bit confused about this line as it mixes the dimension with the label... maybe an explanatory comment would be useful here?

Reply:
Hm, yes this is confusing. Basically, the column naming convention is `0_x, 1_x, 2_x, ... 0_y, 1_y, 2_y, ...`. The number before the underscore indicates which array dimension it belongs to (and the `x`, `y`, etc. are how different origin dataframes are distringuished in the merge with an outer join). We need to find all the columns starting with the same number (array dimension), and then for each row (label) we find the union of those slices.

pseudocode explanation
```
for each array dimension:
    for each label, pick out the values belonging to that array dimension
        and combine those slices
```

I'm not sure I've explained this well, so asking for further clarification or suggesting other improvements would be welcome.